### PR TITLE
Adds a popup with some search tips

### LINKED ISSUE #308: Search tips
Given that we're enabling SOLR search syntax (#306) we could help power users by providing some pointers on using it for advanced search.

Near the search box we could have a link, e.g. "?", "Advanced" or "Advanced Search". That displays some really quick help. I suggest something along these lines:

* `"armed forces" expenses` - search for exact phrases by putting quote marks around them
* `primary school -care` `age NOT "cultural identity"` - exclude words or quoted phrases by using a minus sign or NOT
* `primary OR secondary school` - use 'OR' to allow alternatives (the default is to 'AND' all of the terms)
* "publisher: cabinet-office res_format: CSV" - search only in a particular field. (Lets list all the useful fields from our SOLR schema)

For more detail of the syntax, consult:
* https://lucene.apache.org/core/4_0_0/queryparser/org/apache/lucene/queryparser/classic/package-summary.html

For other tips on searching see:
* http://guidance.data.gov.uk/finding_data.html
* http://guidance.data.gov.uk/api.html#searching

### PR DESCRIPTION
**Pull request by rossjones**
Fixes #308

### CODE REVIEW

File: ckanext/dgu/theme/templates/_dgu_jinja_util.html [hunk 1]
```diff
@@ -117,6 +117,39 @@
...
+            Search Tips
+          </a>
+
+          <div class="panel panel-default dropdown-menu" style="width: 500px; margin-left: -1px;">
+            <div class="panel-heading" style="margin-top: -10px;"><strong>Search Tips</strong></div>
+            <div class="panel-body">
+
+                <ul>
+                  <li><span class='search-terms'>"armed forces" expenses</span> - search for exact phrases by putting quote marks around them</li>
+                  <li><span class='search-terms'>primary school -care age NOT "cultural identity"</span> - exclude words or quoted phrases by using a minus sign or NOT</li>
```
Comment (davidread) [inline ckanext/dgu/theme/templates/_dgu_jinja_util.html]: these are two separate examples. You could put `age NOT "cultural identity"` in a separate span or simply get rid of it.

File: ckanext/dgu/theme/templates/_dgu_jinja_util.html [hunk 2]
```diff
@@ -117,6 +117,39 @@
...
+                  <li><span class='search-terms'>primary OR secondary school</span> - use 'OR' to allow alternatives (the default is to 'AND' all of the terms)</li>
+                  <li><span class='search-terms'>publisher:cabinet-office res_format:CSV</span> - search only in a particular field. A list of available fields is shown in the <a href="http://guidance.data.gov.uk/finding_data.html" target="_blank">Finding Data guidance</a></li>
+                </ul>
+
+                <p>For more detail of the syntax, consult the <a href="https://lucene.apache.org/core/4_0_0/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#minitoc-area" target="_blank">Lucene documentation</a></p>
+
+                <p>For other tips on searching see:</p>
+                <ul>
+                    <li><a  target="_blank" href="http://guidance.data.gov.uk/finding_data.html">http://guidance.data.gov.uk/finding_data.html</a></li>
+                    <li><a  target="_blank" href="http://guidance.data.gov.uk/api.html#searching">http://guidance.data.gov.uk/api.html#searching</a></li>
```
Comment (davidread) [inline ckanext/dgu/theme/templates/_dgu_jinja_util.html]: I suggest you remove this one now

File: ckanext/dgu/theme/templates/_dgu_jinja_util.html [hunk 3]
```diff
@@ -117,6 +117,39 @@
...
+                  <li><span class='search-terms'>primary school -care age</span> - exclude words or quoted phrases by using a minus sign or NOT</li>
+                  <li><span class='search-terms'>primary OR secondary school</span> - use 'OR' to allow alternatives (the default is to 'AND' all of the terms)</li>
+                  <li><span class='search-terms'>publisher:cabinet-office res_format:CSV</span> - search only in a particular field. A list of available fields is shown in the <a href="http://guidance.data.gov.uk/finding_data.html" target="_blank">Finding Data guidance</a></li>
+                </ul>
+
+                <p>For more detail of the syntax, consult the <a href="https://lucene.apache.org/core/4_0_0/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#minitoc-area" target="_blank">Lucene documentation</a></p>
+
+                <p>For other tips on searching see:</p>
+                <ul>
+                    <li><a  target="_blank" href="http://guidance.data.gov.uk/finding_data.html">http://guidance.data.gov.uk/finding_data.html</a></li>
```
Comment (davidread) [inline ckanext/dgu/theme/templates/_dgu_jinja_util.html]: Do you want to add ``#keyword-search-tips``?

File: ckanext/dgu/theme/templates/_dgu_jinja_util.html [hunk 4]
```diff
@@ -117,6 +117,39 @@
...
+            <div class="panel-body">
+
+                <ul>
+                  <li><span class='search-terms'>"armed forces" expenses</span> - search for exact phrases by putting quote marks around them</li>
+                  <li><span class='search-terms'>primary school -care age</span> - exclude words or quoted phrases by using a minus sign or NOT</li>
+                  <li><span class='search-terms'>primary OR secondary school</span> - use 'OR' to allow alternatives (the default is to 'AND' all of the terms)</li>
+                  <li><span class='search-terms'>publisher:cabinet-office res_format:CSV</span> - search only in a particular field. A list of available fields is shown in the <a href="http://guidance.data.gov.uk/finding_data.html" target="_blank">Finding Data guidance</a></li>
+                </ul>
+
+                <p>For more detail of the syntax, consult the <a href="https://lucene.apache.org/core/4_0_0/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#minitoc-area" target="_blank">Lucene documentation</a></p>
```
Comment (davidread) [inline ckanext/dgu/theme/templates/_dgu_jinja_util.html]: This can probably go, now it's in the guidance page

Comment (rossjones) [inline ckanext/dgu/theme/templates/_dgu_jinja_util.html]: Might be worth leaving as a shortcut to visiting another site.

Comment (davidread) [inline ckanext/dgu/theme/templates/_dgu_jinja_util.html]: We have assembled a hierarchy of content - the pop-up has the most important stuff, the guidance has the next most important, and the lucene docs fills in the gaps for the real experts. I don't think the lucene docs should have the prominance of being on the pop-up.

**Pull request merged**